execution/executor-benchmark/src/transaction_executor.rs

```diff
@@ -135,5 +139,11 @@ where
                 .commit_blocks(vec![block_id], ledger_info_with_sigs)
                 .unwrap();
         }
+        self.parent_block_id = block_id;
+
+        if let Some(tx) = &self.maybe_exe_fin_sender {
+            // We are in partition-then-execute mode. Notify the partitioning stage to pick up the next batch.
```

Comment:
Hmm this is a bit convoluted, instead of forcing serialization across multiple threads using this approach, can you think of a cleaner approach? May be if async partitioning is disabled, in the same thread, we first call `partition_block` and then `execute_block`. If it is enabled, then we spawn two separate threads for them?

Reply:
@sitalkedia your example sounds like maintaining 2 pipeline architectures. But I had the impression that the existing pattern is to keep 1 pipeline architecture and use channels to fine-grained control the behaviors (e.g. [here](https://github.com/aptos-labs/aptos-core/blob/65f979aa0bf788ecd84e02399e1a70b7608d1bc9/execution/executor-benchmark/src/pipeline.rs#L65)).
@msmouse @grao1991 your thoughts?

IMO adding a new pipeline stage and an optional sync channel is anyway simpler than keeping 2 pipeline architectures. Maybe I just didn't document well?

Reply:
@zjma - Sorry I didn't make it clearer. The current pipeline architecture is fine, you don't need to change that (except for removing the fin channel). Unless I am missing something, we can get rid of the fin channel and simplify this by doing something like this - 


```

fn partition_block () {

let mut partitioning_stage = BlockPartitioningStage::new(
                    executable_block_sender,
                    maybe_exe_fin_receiver,
                    num_partitioner_shards,
                );
                while let Ok(txns) = raw_block_receiver.recv() {
                    partitioning_stage.process(txns);
                }
            }
   }
   
   fn execute_block(). -> {
   
   start_execution_rx.map(|rx| rx.recv());
                let mut exe = TransactionExecutor::new(
                    executor_1,
                    parent_block_id,
                    version,
                    Some(commit_sender),
                    config.allow_discards,
                    config.allow_aborts,
                    maybe_exe_fin_sender,
                );
                let start_time = Instant::now();
                let mut executed = 0;
                let start_gas = TXN_GAS_USAGE.get_sample_sum();
                while let Ok(msg) = executable_block_receiver.recv() {
                    let ExecuteBlockMessage {
                        current_block_start_time,
                        partition_time,
                        block,
                    } = msg;
                    let block_size = block.transactions.num_transactions();
                    info!("Received block of size {:?} to execute", block_size);
                    executed += block_size;
                    exe.execute_block(current_block_start_time, partition_time, block);
                    info!("Finished executing block");
                }
                
          }
          
          
          If async_partitioning {
          
             thread.spawn(partioning_block).. 
             thread.spawn(execute_block)...
           } else {
              thread.spawn( {
                partition_block()
                execute_block()
            }
```

Reply:
fixed